spec/github_helper_spec.rb

```diff
@@ -171,4 +138,137 @@ def mock_comment(body: '<!-- REUSE_ID: test-id --> Test', user_id: 1234)
       instance_double('Comment', id: 1234, body: body, user: instance_double('User', id: user_id))
     end
   end
+
+  describe 'get_milestone' do
+    let(:test_repo) { 'repo-test/project-test' }
+    let(:release_name) { '10.0' }
+    let(:client) do
+      instance_double(
+        Octokit::Client,
+        list_milestones: []
+      )
+    end
+
+    before do
+      allow(described_class).to receive(:client).and_return(client)
+    end
+
+    it 'receives the correct repository' do
+      expect(client).to receive(:list_milestones).with(test_repo)
+      described_class.get_milestone(test_repo, release_name)
+    end
+
+    it 'returns nil when no milestone exists' do
+      expect(described_class.get_milestone(test_repo, release_name)).to be_nil
+    end
+
+    it 'returns milestone from repo' do
+      allow(client).to receive(:list_milestones).and_return([{ title: 'release/10.0' }])
+      expect(described_class.get_milestone(test_repo, release_name)).to be_nil
+    end
```

Comment:
I'm confused here, because the description of the test says that you are testing if you return the expected milestone from the repo, but your test makes `list_milestones` return one milestone (with a title), yet your expectation is for `get_milestone` to return `nil` (and not a milestone).

I think it's the correct behavior of `get_milestone` here, because the second parameter of `get_milestone` is expected to be used to find the first milestone starting with that text, which a milestone with a title of `'release/10.0'` doesn't. But still:
 - The name of this test is confusing because it suggests that you're testing the case of a milestone being found, while in practice you `expect` the returned value to be `nil` when there's no matching milestone
 - And you actually don't have a test for when `get_milestone` _does_ find a milestone matching the 2nd parameter amongst the list of all the milestones returned by the API.

_PS: In addition, this unit test reveals how unclear the call site of `get_milestone` can be without named parameters (I had to look at the implementation of `get_milestone` to know what the 2nd parameter was used for, and we don't have any YARD documentation on the method either. This could be the occasion to update this `get_milestone` method from the helper (and its call sites) to use named parameters; though that could be done in a separate PR, since that's not directly related to the change with `GITHUB_TOKEN` but more of a tangent improvement idea_

Reply:
>I'm confused here, because the description of the test says that you are testing if you return the expected milestone from the repo, but your test makes list_milestones return one milestone (with a title), yet your expectation is for get_milestone to return nil (and not a milestone).

Sorry for that, was a classic case of CopyAndPasteException 😅 . I'll sort this out and improve the tests to cover the other cases that this method has. Thanks for pointing this out.

> PS: In addition, this unit test reveals how unclear the call site of get_milestone can be without named parameters (I had to look at the implementation of get_milestone to know what the 2nd parameter was used for, and we don't have any YARD documentation on the method either... 

Indeed! This method is so confusing! The `release` param does not state very well that you are searching for a milestone that starts with that name, also you receive only the last one of the milestones that match the name. 

Is very weird to understand that to get a milestone for their number, you should use `get_last_milestone`, and if you wanna get a milestone by name you should use `get_milestone`.

> ...This could be the occasion to update this get_milestone method from the helper (and its call sites) to use named parameters; though that could be done in a separate PR, since that's not directly related to the change with GITHUB_TOKEN but more of a tangent improvement idea

Yeah, I agree! Maybe we can create an issue for registering that? I hope I'm still around to sort this out in near future 😜